### PR TITLE
GtkWpfWidget implementation

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

**Comment by Therzok** [comment]
cc @bratsche @slluis

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Workbench.cs [hunk 1]
```diff
@@ -201,6 +201,11 @@ public Document GetDocument (string name)
 					return false;
 				if (toplevel == RootWindow)
 					return true;
+				#if WIN32
+				var wpfWindow = System.Windows.Application.Current.Windows.OfType<System.Windows.Window>().SingleOrDefault (x => x.IsActive);
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Workbench.cs]: We can't properly detect whether the Window is parented to a GtkWpfWindow. Possibly this needs to be changed.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Windows/GtkWPFWidget.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+		{
+			base.OnShown ();
+
+			wpfWindow.Show ();
+			AttachWindow ();
+		}
+
+		void AttachWindow ()
+		{
+			IntPtr gtkWindowPtr = GtkWin32Interop.HWndGet (Ide.IdeApp.Workbench.RootWindow.GdkWindow);
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Windows/GtkWPFWidget.cs]: We're always attaching to the RootWindow. When we need to attach to an arbitrary window, we need to fix this.

**Comment by alanmcgovern** [comment]
/cc @garuma 

Might be worth investigating to kill the Xwt based renderer in the designer. Maintaining only Wpf and XamMac would be nice :)

**Comment by Therzok** [comment]
Proof from https://github.com/mono/monodevelop/tree/windowsToolbar

**Comment by garuma** [comment]
Is it something we could move to Xwt instead through the wrap native widget facility? That would indeed make life easier for us to kill the xwt-based renderer.

Let's see what @bratsche thinks.

**Comment by alanmcgovern** [comment]
This is analogous to `GtkNSView`, so it should be possible to provide as much XWT support for `GtkWpfWidget` as already exists for `GtkNSView`. We (or @bratsche ) can look into it post c6

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DefaultWorkbench.cs [hunk 3]
```diff
@@ -1028,7 +1028,8 @@ public void ResetToolbars ()
 		protected override bool OnConfigureEvent (Gdk.EventConfigure evnt)
 		{
 			SetActiveWidget (Focus);
-			return base.OnConfigureEvent (evnt);
+			base.OnConfigureEvent (evnt);
+			return false;
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DefaultWorkbench.cs]: This was needed for window changes events to be propagated in the event we hook in the `GtkWPFWindow`.

**Comment by bratsche** [comment]
Would it make sense/be possible for GtkWPFWidget to accept a `Control` instead of a `Window`, and then programmatically create a window and set the state (such as disabling window decorations) on it and add the control to it?

**Comment by Therzok** [comment]
That could work, I can work on getting that.

**Pull request closed without merging**